### PR TITLE
ZAPPD-38: Local Folders Integration Branch

### PR DESCRIPTION
**Pull request by yashrajpchavda**
The branch to hold the work done in epic ZAPPD-38. It would be merged to develop once it is ready to replace the existing local folder implementation.

### CODE REVIEW

File: src/apollo/local-batch-link.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class LocalBatchLink extends ApolloLink {
 							this.schema,
 							query,
 							null,
-							options.context || {},
+							operation.getContext() || options.context || {},
```
Comment (yashrajpchavda) [inline src/apollo/local-batch-link.ts]: operation's context was not passed, passing it here

File: package-lock.json [hunk 2]
```diff
@@ -3765,8 +3765,7 @@
         "ansi-regex": {
```
Comment (silentsakky) [inline package-lock.json]: revert package lock changes

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -613,6 +618,30 @@ export class ZimbraBatchClient {
...
+		this.jsonRequest({
+			name: 'GetMsgMetadata',
+			body: {
+				m: {
+					ids: ids.join(',')
+				}
+			}
+		}).then(res => {
+			// logs kept to verify the data when available
+			console.log('metadata res is', res);
```
Comment (silentsakky) [inline src/batch-client/index.ts]: remove console logs

File: package-lock.json (hunk 2 above)
Comment (yashrajpchavda) [reply package-lock.json]: fixed

File: src/batch-client/index.ts (hunk 3 above)
Comment (yashrajpchavda) [reply src/batch-client/index.ts]: fixed

**Pull request merged**